hawtjni-example/src/main/java/test/Example.java

```diff
@@ -88,6 +88,9 @@ public static final void main(String args[]) {
     public static int O_WRONLY;
     @JniField(flags={CONSTANT})
     public static int O_RDWR;
+
+    @JniField(flags={SHARED_PTR})
+    public static int CheckStr;
     
```

Comment:
The added field in the example is not taken into account and does not exercise the code at all.

Reply:
@gnodet thanks for the reviewing the changes
we added SHARED_PTR field in the example and hawtJNI generator is able to produce corresponding JNI for this field.
can you tell what is the expectation from the example?

Reply:
The expectation is that the field is not ignored and ends up being used in the generated code somehow, which is not the case here.  I think that's because it's a static field.

Reply:
I pushed an update to the example that actually uses the shared pointer.  The example does not work because of the missing field.  If you can make it work, I'll gladly merge the PR and perform a release.
  https://github.com/gnodet/hawtjni/tree/ossdev07-shared_ptr_support

Reply:
@gnodet I checked your commit and need some suggestion
currently, CheckStr is defined as follows
```
@JniField(flags={SHARED_PTR})
 private int CheckStr;
````
and I saw, another JniField is defined as -
```
  @JniField(flags={CONSTANT}, accessor="sizeof(struct foo)")
        public static int SIZEOF;
```
So, can we define CheckStr as static field like
```
@JniField(flags={SHARED_PTR})
private static int CheckStr;
```
please suggest ...

Reply:
Static fields are ignored when generating getters because of the `if (ignoreField(field)) continue;` statement at the beginning of the loop in the `generateGetFields` method, line 282-283.